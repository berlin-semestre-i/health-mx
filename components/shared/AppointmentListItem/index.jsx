import React from 'react'
import { Grid } from 'semantic-ui-react'
import styled from 'styled-components'
import Link from 'next/link'
import media from 'styled-media-query'
import Avatar from '../AppointmentAvatar'
import ConsultationInfo from '../AppointmentInfo'
import ValueItem from '../ValueItem'

const AppointmentListItem = ({ appointment }) => (
  <ConsultationRow>
    <AvatarContainer mobile={16} tablet={3} computer={2}>
      <Avatar src={ appointment.profile }/>
    </AvatarContainer>
    <ConsultationInfoContainer mobile={16} tablet={13} computer={14}>
      <Link as="" href=""><a>{ appointment.name }</a></Link>
      <ConsultationInfo>
        <ValueItem keyName="Fecha" value={ appointment.date } />
        <ValueItem keyName="Hora" value={ appointment.time } />
        <ValueItem keyName="Ãšltima cita" value={ appointment.previous } />
      </ConsultationInfo>
    </ConsultationInfoContainer>
  </ConsultationRow>
)

export default AppointmentListItem

const ConsultationRow = styled(Grid)`
  >.column:not(.row):last-child {
    padding-top: 1.5rem;
  }
`
const AvatarContainer = styled(Grid.Column)`
  text-align: center;

  ${media.between('medium','1025px')`
    [class*="two wide computer"]&&&.column {
      width: 18.75%!important;
    }
  `}

  ${media.lessThan('medium')`
    &&&.column:not(.row) {
      padding-bottom: 0;
    }
  `}
`
const ConsultationInfoContainer = styled(Grid.Column)`
  ${media.between('medium','1025px')`
    [class*="fourteen wide computer"]&&&.column {
      width: 81.25%!important;
    }
  `}
`
const Bold = styled.b`
  font-weight: 500;
`