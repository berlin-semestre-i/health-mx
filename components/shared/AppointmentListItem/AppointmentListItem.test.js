import React from 'react'
import { shallow } from 'enzyme'
import AppointmentListItem from './index'

describe('AppointmentListItem', () => {
  it('renders', () => {
    const component = shallow(<AppointmentListItem appointment="{}" />)
    expect(component).toMatchSnapshot()
  })
})