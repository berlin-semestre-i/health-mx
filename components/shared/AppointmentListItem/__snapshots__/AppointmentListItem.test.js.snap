// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentListItem renders 1`] = `
<AppointmentListItem__ConsultationRow>
  <AppointmentListItem__AvatarContainer
    computer={2}
    mobile={16}
    tablet={3}
  >
    <AppointmentListItem__Avatar />
  </AppointmentListItem__AvatarContainer>
  <AppointmentListItem__ConsultationInfoContainer
    computer={14}
    mobile={16}
    tablet={13}
  >
    <Link
      as=""
      href=""
    >
      <a />
    </Link>
    <AppointmentListItem__ConsultationInfo>
      <div>
        <AppointmentListItem__Bold>
          Fecha: 
        </AppointmentListItem__Bold>
         
      </div>
      <div>
        <AppointmentListItem__Bold>
          Hora: 
        </AppointmentListItem__Bold>
         
      </div>
      <div>
        <AppointmentListItem__Bold>
          Ãšltima cita: 
        </AppointmentListItem__Bold>
         
      </div>
    </AppointmentListItem__ConsultationInfo>
  </AppointmentListItem__ConsultationInfoContainer>
</AppointmentListItem__ConsultationRow>
`;
